### PR TITLE
WIP use Arc<str> instead of Arc<String> 

### PR DESCRIPTION
**Pull request by olexiyb**
This should fix one of clippy errors we started to see with latest stable 1.48.0
https://rust-lang.github.io/rust-clippy/master/index.html#rc_buffer

### CODE REVIEW

File: src/actor/uri.rs [hunk 1]
```diff
@@ -4,11 +4,11 @@ use std::{
     sync::Arc,
 };
 
-pub struct ActorPath(Arc<String>);
+pub struct ActorPath(Arc<str>);
```
Comment (igalic) [inline src/actor/uri.rs:7]: shouldn't this be `&str`?

**Review by igalic — COMMENTED** [review]
🙋🏻‍♀️

File: src/actor/uri.rs (hunk 1 above)
Comment (olexiyb) [reply src/actor/uri.rs:7]: This will be way harder to do as you need to add a lifetime parameter and it should be &'a str.

**Comment by olexiyb** [comment]
I would like to wait for #145 merge. I may need to resolve conflict related to this merge

**Comment by nothingismagick** [comment]
yeah, that was what i was seeing on clippy nightly. i also wonder about that toString recursion issue...

**Comment by olexiyb** [comment]
> yeah, that was what i was seeing on clippy nightly. i also wonder about that toString recursion issue...

I plan to propose a pull request after the merge of this one

**Comment by nothingismagick** [comment]
I guess we have to wait a bit, switching CI is never a simple choice.

**Pull request merged**